julia-mode.el

```diff
@@ -459,7 +459,20 @@ Do not move back beyond MIN."
     (and pos
 	 (progn
 	   (goto-char pos)
-	   (+ julia-indent-offset (current-indentation))))))
+	   (+ julia-indent-offset (julia-block-open-indentation))))))
+
+(defun julia-block-open-indentation ()
+  "Get the current indentation or the start of a parenthetical block."
+  (save-excursion
+    ;; narrow to one line to only search syntax on that line
+    (narrow-to-region (line-beginning-position) (line-end-position))
+    (prog1
+        (condition-case nil
+            (progn
+              (backward-up-list)
+              (1+ (current-column)))
+          (error (current-indentation)))
+      (widen))))
```

Comment:
instead of `(widen)`, perhaps wrap the narrowing in a `(save-restriction ...)` block?